source/union-to-tuple.d.ts

```diff
@@ -0,0 +1,41 @@
+import {UnionToIntersection} from './union-to-intersection';
+
+/**
+Convert an explicit union type to a tuple.
+
+The element order of the tuple and of the union will be the same. Because of that, you should not use this if the union is computed instead of manually declared, because the TypeScript compiler does not guarantee that the order of such union elements in that case.
+
+Inspired by [this issue in the TypeScript repo](https://github.com/microsoft/TypeScript/issues/13298).
+
+Use-case: You need to do validation but you don't have an enum, only an union type. Similar to the [`Extract` example in the readme](https://typescript-play.js.org/?target=6#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXzSwEdkQBJYACgEoAueVZAWwCMQYBuAKDDwGcM8MgBF4AXngBlAJ6scESgHIRi6ty5ZUGdoihgEABXZ888AN5d48ANoiAuvUat23K6ihMQ9ATE0BzV3goPy8GZjZOLgBfLi4Aejj4AEEICBwAdz54MAALKFQQ+BxEeAAHY1NgKAwoIKy0grr4DByEUpgccpgMaXgAaxBerCzi+B9-ZulygDouFHRsU1z8kKMYE1RhaqgAHkt4AHkWACt4EAAPbVRgLLWNgBp9gGlBs8uQa6yAUUuYPQwdgNpKM7nh7mMML4CgA+R5WABqUAgpDeVxuhxO1he0jsXGh8EoOBO9COx3BQPo2PBADckaR6IjkSA6PBqTgsMBzPsicdrEC7OJWXSQNwYvFEgAVTS9JLXODpeDpKBZFg4GCoWa8VACIJykAKiQWKy2YQOAioYikCg0OEMDyhRSy4DyxS24KhAAMjyi6gS8AAwjh5OD0iBFHAkJoEOksC1mnkMJq8gUQKDNttKPlnfrwYp3J5XfBHXqoKpfYkAOI4ansTxaeDADmoRSCCBYAbxhC6TDx6rwYHIRX5bScjA4bLJwoDmDwDkfbA9JMrVMVdM1TN69LgkTgwgkchUahqIA), but in that case the `changePersonData` function does not validate the `key` parameter at runtime, making it only safe inside TypeScript. Using a tuple and `Array#includes` for validation gives us runtime safety for that kind of function.
+
+@example
+```
+type Union = 'A' | 'B';
+
+const tuple: UnionToTuple<Union> = ['A', 'B'];
+```
+
+@example
+```
+// This type may come from a library that you are using.
+type ActionsInSomeLibrary = 'create' | 'read' | 'update' | 'delete' | 'aggregate';
+
+type AllowedActions = Exclude<ActionsInSomeLibrary, 'aggregate'>;
+
+const actions: UnionToTuple<AllowedActions> = ['create', 'read', 'update', 'delete'];
+
+actions.includes(requestedAction) // Validate
```

Comment:
What benefit does this provide over `AllowedActions[]`? It's likely just me missing something, but I'm thinking, if I'm missing something then someone else can as well, so then we should clarify it 🙂 

```typescript
const actions: AllowedActions[] = ['create', 'read', 'update', 'delete'];
```

I get the impression that `actions.includes(requestedAction)` will work better when the type of `actions` is `UnionToTuple<AllowedActions>` rather than `AllowedActions[]`? But I can't reproduce that: [TypeScript Playground example](https://www.typescriptlang.org/play?#code/PQKhCgGEHsDsDcCmAnALgAgIboK6wJZzqoCeADosdFrOvrKigM6IDGqhtpFuT9A5ugDaAE3xNUyfACMcHJOlZwxHOJgA2xcoiYBdABQALVKjJMAXMGAB3WwDpuO1lLKp1mWPzvRk-YCOhWJmBDDxFpaGgAa2BkRHVETBYAWlhoRmDHJmd8V2SAJmSADjtjAFt1AGIxCSlZeURkpVgVTg1krIBKO3BwAElYJjJ8OJF0aRJhVENxdABlVExWKPQAeSRkADN1aGsaJmsUA2NTCysJJajoDe3duyUy4ExgAFYABgBmAHYX-KKANmA+QAjF8QS8Pt1egABRAAD0wZTICXAAAN0eB8EifBgAN4AVQIcAAKtABoxkCx2JwAL7oTbIaBldAAckcyU2OlQLIA3L1HOhCZx0ABedC46aIfSdczoeDQfAiOkAH3F-DimFQ+kwvlltQEMrlCqV6FVuKcmAosoi0ASHhpfPAAvJzDYqloYqFJLJDFd1LgAB4vbAAHx8qwikPiyXS2XyxU89DqxJanX8PWSA1x42Ji1W8aRO2wB1ojHgACC6DKPkoluR+FYmGkCXQ8MRyMo1hmrEMimgOHUYzKmCilHwqCY6GsmEm9FQ1BI-eQ6HUMmQOsmShElE2tp21iYPXAsIRSJR6NRmOxaHFwdJLspbtp9MZzLZ2g5XN5vVY7iYk5gMphxaVZYEoXFwAASAeYCRAscUoMgzBgVlaVRSjPBt02ehEBEAAaRDpBQ9A0MjXAWkQbCwPwqCaXAOjwF-JIAKZWDiWsagIOg1iwngrikPyVC03LDMpE8Tp0PIrCcJoyDIOkQSSLTAAhUSDUkzDKJkgjILohjmgkcjhTFIQwL2QDYNAqVOjw9AzPQCywnY6BpV0OxhzIfR6AuWBWEoMjvMWXzEHuHiWiYTo+QFYNRS0ChoE2Iy4HDYAyNxZDYyNBNxmBTL4xNM1MHybVfBE9B9XE7NsoUkr+FU8rM0qrKlX5bR0AfKl3Viu8fQpTrOCDIlQxStKMsNfLEyIvKcywYrhLUpqJvGObfHqir+HG406IvcAQGAcB4TIHE4soHqOqfQNgyjMV9Cgqx0FReFGHC8iojSaxYFROhJw0acSEnfgFU8KhxkoSVFCSWsWm+3hcJB5oNgwSU7uAB7gy+udqGwURxEzep8AUZpWjUdQUZO44TDMSwbHsLIclcdxPG8Xx-ECYJQhaG0YjiO0UjSDJgDplxUAKYpSlQCpqlxuo5AJxoifHNp1A6bQIp6SCYqexAXrwN7dlgRD7uJQxKDwYUBVmHAWDGJJiBN9A4HUSY0xwMptaRrH6Twf1aD4YK7dNobDdRhKvd8rqXbdhhJ1mDw6F9R8ffVyCAH4SJqPG5Fw4NZWDCSyPy4P0GN2YONgNl0FCMgKFgZPZTAjYi4GTll3B1Fzp9r6BT9vzi+0OZ6YwOIyDiFho4D9Ajr4d0NCLyP3cnUOzbgRfEs2b33R+n7aDnP0utD8Hl9rqCJK1l79H0N3fFwjv3VlegW-ahP+rgfOo3yzpELT2-OEQ+vEA2I6UAEAYAIBQBgOOh1VysHHElLgbV5xYGIDgDsPRjaUHiIgKOGAfDbmXAfe2qAUEtjCA7RKh8hpTnwOoTQ0gwb2yYIiEK6BlJsEwFbSgBDNS2UXDgcqhh+yDjsukWGdtZj4HIfbI+MMHhkCzmMQKiQxih2AjgDQTt0DbiYqMWydDGwcInsSfug8+xImoSgTR0AdDCIwPwNR65fT0M1BPXBFjQ5MBwD2OBrYEjYJjlwUIGBGwsB6AMIYIw4YTCmDMGO-4cBjgCZQIxFAB7C3QMPaAFNTjU34OOQwOBpChUeGUBsjImAJVQMAZJiBUm5EqeIDxOhgDAg+PkAAnEUKE+IUjBMQLKAAmv2OyiA4aIICHKDQipNTCnqOgXhljy4YFCAoSBsBXa2Uds7Wg0jHB2HmFiahOoQbgyEKiAAonCSQSxUBfTbGeBJE8NQiDdlkqmVghZ1OSMiGcdgABWh4fB+BTosa+qART-EqFuRAwBID9PxNYOE-AABy-BDD8GJPgcsZzywAC9yy+FWHigAYtYAAUv0gAEqgAAGjiuY1gzkiCiAARWUuS8skB+BnOgOWeJAA1AAWuWAA6tYSAABZZl-TlI4HLAAaQACIKusAAcVYOKoo4q6pEoACzlmpZ4ZS6hyykv+NkM5cx+AUr6AAJXwP8UgLwBX4hVQEfAaLsXKWpQKoovryxIpeCIHVRR-UKl5fAfEmp8gfDlf8d1kq2nlmJGct4ykcV8o+IDAACiQIoKqBW-IFasAAMnVTYxb8B6sQL835eqzlnL6JAZS1hywiBxS8HV4ryzlmLXKolzLmUAGplJwjOYgbtFL+nAhRXK8sHE+hypIG8dUyAdUKuUiQM5+IyD8FYKwHd4rMDUv4OWTAcJWHIEgDi-Aw62nJAFQOEg-AFX9iJRSm1TB8TAhxUUKIcrxX9LOcCG1oQACOJ6KVRFQHqyDwqOXQexeWLN0g+U2v4MW4twqUXKTIG0-gKr1DKSYEUHAzLMD-BIOWfE+IcD9Kzcy6wIhkVkH-V8WAhgvhlHFeK4tOrOXlkHTal4cHlKgfxOWfgZAFWID5XCHAhg4SrGBCbN4-zqUqsMEUblqxlKrDaZAVYcIPjKWZVm6A+Qs3KXLAq5YmAbX-D6L8qIcxyz-Es6B4k-AmDirTVmvgrARDIDOZAL4qxSXCupXMZlSLrD9PgESs5J6+XEjmG00lxbqWrAVRQbLcrlICqJfwHVKqYDCswEUPlHK+ikpIFEeV+BmXCsXfkfpzLVjlkIP0-AUROVvFWGc8VCqSCGBtXMEga6SBwjmPkHVcrDDdvFb8kg9r+BzBDXO35hgXhZbePgZSb7yxRFJdAM5qwohMEgMCMosAojitJaBoo-B8TMsVUikwcqs3qFgAyWLZAPikpeNSzYykKXUtA9AOVZBNj9Lq6sPoOqPBMGJAiRACrrPVlG5AJt-TyzSDhIYSA-xiQKrhP8ZAsXrU2upS8aQA9fnlh1dIYtpKOIKrKMqhVURNjSHLGl8VyA+Xir5SIcVwJiRIv+G00tURPPWH4FEHskbDCgb6OWGy4w5Dxzts43pE9UQ9g8PwRAWbmBwAVZqTAX117h2FAEax-MJmrhEJqehlBUSjhIF9MgOomEUiwEPPAHA3a2WHFEAQdAcGwA0Ywzk8c+Dbj7ikweezunh+wEQjsNAxionxeuEglR6C-hwNuJgVufCO6mV1XJSBJxW3SYHrElAY+IFIPScv0xnFh+hqHa3Sdjx3I7KWS80VKFihZOWFkppWTKW-AZJGxC+mCiGqSYkC-BqcGusIcfLJbIshn7oPkO0TztnPBiI2MSTpVhnKYncr4kGrmkOuZAkwO8YHmTqU2fsvBOjauWH2TABg5gmREAK1H8NxYoWRnAUxEBJ9VQWQnlYDWQUFndGBECWRtwEhUCp8WRMB+Bkx+AXdvwBRywaFdhcI-9N5YoLki9twAwKDOAADYAgC3ZQCn8SBd9cD8DCCwxGIV4IF-9c5l9oBV8Ow6DSDDgRB6CV5N8hBICNRUDd8EDd9kDCDd8MCW8YCD9wBrkGC7BC91Bi8dB9A4hQN4kJByCfYJJ7o+VJkUDEAh9dp9pDpjoR9OAV818rpRRwB0AfCl83Deo94BpvDfCQj7oFVpYZBZZCZlBFYSZiApBlgegQjQjUY5gRkJ5ykcBkBe5dxW57ZUQzpn4LpPpL9ciqwaxNEW9MBqFDxgjki-CiAz44JXp3oDZ6ifC059AM4ZZGARBBDOA35mo6jkj-5G4QiowmjJwL4r4jc+j45H4f5X5JIP5hiUji4HEmBciygpjt9sCZ8JJMZhAhAhBdBbIAAicsM4049AM45SK43RTXEvSQTxIhUeCeHUPPJgVY3wtOJ5TZYQOwQEnqEQhIAMaggw2g4MWyRY0MEMaEoon2XQb4nw2UE4oBVGEuScAUYcTcYAl8Jke-NccA1-OZIZD-XgAQH-HgKQwYQA4A1g8AsfKAwg7AhA7AlQrAuA9Qzk1kTguIAg1Ax0Yg8QiwygsUcEwwug--Oklgok5-DgvA-k7gx0OfLAAQho2Adw0Qkg-cUUhgmQuQ6AnfVkJQpAsgOw409AzBVArQ8Ae6EgikWAaZFZauRkJYXsawPJfsDAYEtfEMXgwYfgzeRSHUsgyQ--E42KWQ5khQk0pRS0jkmAtQ60zQ9E9AUlK2DAOhB2DYDRX8RIZcJsb0ieeADQeJAMwyEwsw3omk2UQ0lkpk+QmAtMlVFvLEtqM4m0IsM4g6OEI6G8VU3cagMUHQlePQ3yCEowqsrkPU1+O01GEg8pJMNsy-TswsRIWAHs5wgcvgtvczWKUcwYfIccmgqcxAUwmc8M90ToIAA)

Also, I think that `requestedAction` needs to be defined in this example, else one doesn't know what type that one has at all, and that seems to be vital to the example. So add something like `const requestedAction = 'create';`?

Reply:
> What benefit does this provide over `AllowedActions[]`?

With just `AllowedActions[]` you are allowed to do `const actions3: AllowedActions[] = [];` and `const actions2: AllowedActions[] = ['create', 'create', 'create'];` which are not be what is actually intended. Do you think this should be talked about in the description?